common/src/main/scala/workers/WebWorkerF.scala

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2016-2022 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package workers
+
+import cats.effect.Async
+import cats.effect.Sync
+import cats.effect.std.Dispatcher
+import fs2.Stream
+import fs2.concurrent.Channel
+import org.scalajs.dom
+
+import scala.scalajs.js
+
+/**
+ * WebWorker abstraction running on F. it is possible to post messages and get a stream of events
+ * from the web worker
+ */
+trait WebWorkerF[F[_]] {
+
+  /**
+   * Post a message on effect F
+   */
+  def postMessage(message: js.Any): F[Unit]
+
+  /**
+   * Terminate the web worker
+   */
+  def terminate: F[Unit]
+
+  /**
+   * Streams of events from the worker
+   */
+  def stream: Stream[F, dom.MessageEvent]
+}
+
+object WebWorkerF {
+
+  def apply[F[_]: Async](worker: dom.Worker) = new WebWorkerF[F] {
+    def postMessage(message: js.Any): F[Unit] = Sync[F].delay(worker.postMessage(message))
+
+    def terminate: F[Unit] = Sync[F].delay(worker.terminate())
+
+    def stream: Stream[F, dom.MessageEvent] =
```

Comment:
We can pass the `Dispather` implicitly and reuse the one we already have in explore's context.

Reply:
Calling `stream` multiple times will allocate multiple `Channel`s and register multiple `onmessage` callbacks. Is that what we want?

Otherwise we can set up the `Channel` and register a callback in the enclosing `WebWorkerF` constructor.

Reply:
Would it be enough to convert this to a lazy val?

Reply:
I think that would work, yes.